File: cardmarket-frontend/src/api/sealed.ts
import { apiGet } from "./client";

export interface SealedContentItem {
  id: number;
  product_id: number;
  component_type: string;
  qty: number;
  linked_product_id: number | null;
  linked_product_name: string | null;
  product_name?: string | null;
}

export interface SealedContentsResponse {
  product_id: number;
  items: SealedContentItem[];
}

export async function fetchSealedContents(productId: number): Promise<SealedContentsResponse> {
  return apiGet<SealedContentsResponse>(`/sealed-contents/${productId}`);
}

File: cardmarket-backend/app/schemas/sealed.py
from pydantic import BaseModel
from typing import Optional, List


class SealedContentItem(BaseModel):
    id: int
    product_id: int
    component_type: str
    qty: int
    linked_product_id: Optional[int] = None
    linked_product_name: Optional[str] = None
    product_name: Optional[str] = None  # nur für List-Endpoint


class SealedContentsResponse(BaseModel):
    product_id: int
    items: List[SealedContentItem]
